shared/tests/utility/math/filter/PFVanDerPolModel.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUBots <nubots@nubots.net>
+ */
+
+#ifndef SHARED_TESTS_PFVANDERPOLMODEL_H
+#define SHARED_TESTS_PFVANDERPOLMODEL_H
+
+#include <Eigen/Core>
+
+namespace shared {
+namespace tests {
+
+    template <typename Scalar>
+    class PFVanDerPolModel {
```

Comment:
Isn't both this model and the UKF model the same?

Reply:
They are, apart from the differences between models needed for PF and UKF

Reply:
Maybe just have a single model then, I think PF just has extra values

Reply:
This comes down to how we want to handle measurements of a single value.

ParticleFilter implementation happy to use `Eigen::Matrix<Scalar, 1, 1>`, UKF is not and I had to make a reimplement `measure` to accept `Scalar` values (I assume you are fine with this implementation?).

We can either reimplement `measure` in ParticleFilter to accept `Scalar` values, try and fix the issues in UKF `measure` for `Eigen::Matrix<Scalar, 1, 1>`, or leave things as they are and have the two models here.

Reply:
The idea is that both models are compatible with each other so you can swap between PF and UKF mostly transparently. Honestly, I'm not sure the rouge count etc belongs in the model, Maybe it should be added into the PF itself.

Reply:
This may be true, but the filters don't handle scalar measurements in the same way